app/presenters/tutorial_view_step_presenter.rb

```diff
@@ -0,0 +1,52 @@
+require 'tutorial_view_chapter_presenter'
+
+class StepPresenter
+  def initialize(step, chapter)
+    @step = step
+    @step_order = nil
+    @chapter = chapter
+  end
+
+  def step_ordering()
+    ordering = {}
+    counter = 0
+    @chapter.steps.each do |step|
+      if step.warning?
+        ordering[step] = "WARNING"
+      else
+        counter += 1
+        ordering[step] = "STEP: #{counter}"
+      end
+    end
+    return ordering
+  end
+
+  def order_check
+    order = step_ordering()
+    order.each do |key, value|
+      if @step == key
```

Comment:
Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.